libcloud/compute/drivers/outscale.py

```diff
@@ -0,0 +1,1109 @@
+        :type       description: ``str``
+
+        :param      image_name: A unique name for the new OMI.
+        :type       image_name: ``str``
+
+        :param      block_device_mapping: One or more block device mappings.
+        :type       block_device_mapping: ``dict``
+
+        :param      no_reboot: If false, the VM shuts down before creating
+        the OMI and then reboots.
+        If true, the VM does not.
+        :type       no_reboot: ``bool``
+
+        :param      root_device_name: The name of the root device.
+        :type       root_device_name: ``str``
+
+        :param      source_region_name: The name of the source Region,
+        which must be the same
+        as the Region of your account.
+        :type       source_region_name: ``str``
+
+        :param      file_location: The pre-signed URL of the OMI manifest
+        file, or the full path to the OMI stored in
+        an OSU bucket. If you specify this parameter, a copy of the OMI is
+        created in your account.
+        :type       file_location: ``str``
+
+        :param      dry_run: If true, checks whether you have the required
+        permissions to perform the action.
+        :type       dry_run: ``bool``
+
+        :return: the created image
+        :rtype: ``dict``
+        """
+        data = {
+            "DryRun": dry_run,
+            "NoReboot": no_reboot,
+        }
+        if block_device_mapping is not None:
+            data.update({"BlockDeviceMappings": block_device_mapping})
+        if image_name is not None:
+            data.update({"ImageName": image_name})
+        if description is not None:
+            data.update({"Description": description})
+        if vm_id is not None:
+            data.update({"VmId": vm_id})
+        if root_device_name is not None:
+            data.update({"RootDeviceName": root_device_name})
+        if source_region_name is not None:
+            data.update({"SourceRegionName": source_region_name})
+        if file_location is not None:
+            data.update({"FileLocation": file_location})
+        data = json.dumps(data)
+        action = "CreateImage"
+        signer = OSCRequestSignerAlgorithmV4(access_key=self.key,
+                                             access_secret=self.secret,
+                                             version=self.version,
+                                             connection=self.connection)
+        headers = signer.get_request_headers(action=action,
+                                             data=data,
+                                             service_name=self.service_name,
+                                             region=self.region)
+        endpoint = self._get_outscale_endpoint(self.region,
+                                               self.version,
+                                               action)
+        return requests.post(endpoint, data=data, headers=headers)
+
+    def list_images(self, data="{}"):
```

Comment:
Needs to return a list of ``NodeImage`` objects. data is a non-standard argument so it needs to be prefixed with ``ex_``.